crates/ruff_python_formatter/src/expression/expr_bin_op.rs

```diff
@@ -36,68 +36,122 @@ impl FormatNodeRule<ExprBinOp> for FormatExprBinOp {
-            } else {
-                soft_line_break()
-            };
-
-            write!(
-                f,
-                [
-                    left.format(),
-                    before_operator_space,
-                    op.format(),
-                    trailing_comments(operator_comments),
-                ]
-            )?;
-
-            // Format the operator on its own line if the right side has any leading comments.
-            if comments.has_leading_comments(right.as_ref()) {
-                write!(f, [hard_line_break()])?;
-            } else if needs_space {
-                write!(f, [space()])?;
+            BinaryLayout::Default
+        };
+
+        match layout {
+            BinaryLayout::Default => {
+                let comments = f.context().comments().clone();
+                let operator_comments = comments.dangling_comments(item.as_any_node_ref());
+                let needs_space = !is_simple_power_expression(item);
+
+                let before_operator_space = if needs_space {
+                    soft_line_break_or_space()
+                } else {
+                    soft_line_break()
+                };
+
+                write!(
+                    f,
+                    [
+                        left.format(),
+                        before_operator_space,
+                        op.format(),
+                        trailing_comments(operator_comments),
+                    ]
+                )?;
+
+                // Format the operator on its own line if the right side has any leading comments.
+                if comments.has_leading_comments(right.as_ref()) {
+                    write!(f, [hard_line_break()])?;
+                } else if needs_space {
+                    write!(f, [space()])?;
+                }
+
+                write!(f, [group(&right.format())])
             }
 
-            write!(f, [group(&right.format())])
+            BinaryLayout::ExpandLeft => {
+                let left = left.format().memoized();
+                let right = right.format().memoized();
+                write!(
+                    f,
+                    [best_fitting![
+                        // Everything on a single line
+                        format_args![left, space(), op.format(), space(), right],
+                        // Break the left over multiple lines, keep the right flat
+                        format_args![
+                            group(&left).should_expand(true),
+                            space(),
+                            op.format(),
+                            space(),
+                            right
+                        ],
+                        // The content doesn't fit, indent the content and break after the operator.
+                        format_args![
+                            text("("),
+                            block_indent(&format_args![
+                                left,
+                                hard_line_break(),
```

Comment:
This formatting always looks kinda odd to me but i guess python doesn't allow much better